### ISSUE TITLE
Unable to move window using touch screen

### ISSUE THREAD

**Issue by JamesSwift** [johnfactotum/foliate#363]
As expected, when you grab the top bar of foliate with a mouse you can move it around your screen. When using a touch-only device you can't grab the top bar and move the window, unless you disable "auto hide header bar".

**Comment by johnfactotum** [comment]
What happens if you tap on the middle of the page (which should reveal the header bar if you have "tap to turn page" enabled), can you drag it then?

**Comment by JamesSwift** [comment]
I should have been clearer sorry. That's what I was already doing. 
Nothing happens.

**Comment by johnfactotum** [comment]
That's quite strange. It should not matter anyway since whatever's packed in the titlebar area should be draggable. Unfortunately I don't have a touchscreen so I can't test this.

**Comment by JamesSwift** [comment]
We need to get you a touch enabled device 😁️ I'm a big fan of the Acer Switch 11 sw5-173 (4GB RAM, 64GB SSD), you can often get them used in good condition for around £100.  You can put pretty much any distro on them and it works out of the box, although I've found Pop on wayland to be the best experience. Just avoid the similarly named but lower spec acer switch 10. Or maybe we could get pine64 to donate a pinephone?

**Comment by jhakonen** [comment]
I was able to reproduce this with my touchscreen device (eve-v) using Wayland. Couldn't reproduce with Xorg, there the dragging worked fine.

**Comment by kysrpex** [comment]
Can reproduce this as well using Wayland. Dragging with my finger does not work unless "auto hide header bar" is disabled.

**Comment by johnfactotum** [comment]
Would be great if someone could test with the `gtk4` branch. There's no title area anymore, instead the top area is wrapped in a GtkWindowHandle.

**Comment by ivzave** [comment]
I've just checked out `gtk4` branch and compared its behaviour against `master` on touch-enabled device.

First, about `master` branch:

- with "hide header bar" option off moving window works just fine
- with "hide header bar" option on I can't drag window, however I can touch center of window so header bar + footer appear and then I can drag window if I start dragging on any of header bar controls like left TOC button or right buttons. If I start gesture at book title in header bar nothing happens

On `gtk4` branch dragging mostly works. When I start dragging window header bar appears and doesn't disappear after I finish gesture. Touching central part of window doesn't control header+footer appearance unlike `master` branch. Also I can touch bottom part of window so footer appears and I am able to navigate book, however I can't make footer to disappear after that.

Also I have found using touchscreen can break window operation. Buttons stop reacting to touchscreen and even touchpad events and window can't be moved except by keyboard. However I was also able to break this way GNOME Text Editor so it's probably libadwaita or GTK being buggy.

OS: Fedora 37
DE: GNOME 43.3/Wayland
Device: Dell XPS 15 9560
